tests/test_paperqa.py

```diff
@@ -621,14 +623,17 @@ def test_duplicate(stub_data_dir: Path) -> None:
     ), "Unique documents should be hashed as unique"
 
 
-def test_docs_with_custom_embedding(subtests: SubTests, stub_data_dir: Path) -> None:
+@pytest.mark.parametrize("vector_store", [NumpyVectorStore, QdrantVectorStore])
+def test_docs_with_custom_embedding(
```

Comment:
Running this test locally I get this:

```
TypeError: cannot pickle '_thread.RLock' object
```

In the `docs_deep_copy = deepcopy(docs)` line

Can you run tests locally and get them to pass?

Reply:
`QdrantClient` cannot be deep copied. So I've removed the deep copy checks. Since they're not applicable to the how `QdrantVectorStore` works anyway.

Reply:
We need those assertions as they provide coverage on `PaperQAEnvironment`/`GradablePaperQAEnvironment.__deepcopy__`

You'll need to comprehend how `deepcopy`ing of an environment containing a `QdrantVectorStore` would play out. Perhaps you need to implement `QdrantVectorStore.__deepcopy__`

Reply:
> TypeError: cannot pickle '_thread.RLock' object

Should be resolved now.

Will need to figure how to make 

```py
assert docs.texts_index != docs_shallow_copy.texts_index
```
pass without a duplicate `_embeddings_matrix`.

Reply:
https://github.com/Future-House/paper-qa/pull/730/commits/9836e09dec8c5ffa6c44c30e4d4754584503b961

Should do it.